kong/db/schema/metaschema.lua

```diff
@@ -496,12 +584,46 @@ check_field = function(k, field, errors)
   for name, _ in pairs(nested_attributes) do
     if field[name] then
       if type(field[name]) == "table" then
-        check_field(k, field[name], errors)
+        check_field(k, field[name], errors, field)
       else
         errors[k] = meta_errors.TABLE:format(name)
       end
     end
   end
+
+  if field.type == "json"
+    and field.json_schema
+    and field.json_schema.parent_subschema_key
+  then
+    local parent_subschema_key = field.json_schema.parent_subschema_key
+    local found = false
+
+    for i = 1, #parent_schema.fields do
+      local item = parent_schema.fields[i]
+      local parent_field_name = next(item)
+      local parent_field = item[parent_field_name]
+
+      if parent_subschema_key == parent_field_name then
+        if parent_field.type ~= "string" then
+          errors[k] = errors[k] or {}
+          errors[k].json_schema = {
+            parent_subschema_key = meta_errors.JSON_PARENT_KEY_TYPE
+          }
+        end
+        found = true
+        break
+      end
+    end
+
+    if not found then
+      errors[k] = errors[k] or {}
+      errors[k].json_schema = {
+        parent_subschema_key = meta_errors.JSON_PARENT_KEY
+      }
+      return
+    end
+  end
```

Comment:
FYI for the sake of stylistic consistency, this was copy/pasted and modified from the `subschema_key` metaschema validation code, since it serves a similar function.